contracts/crowdsale/emission/AllowanceCrowdsale.sol

```diff
@@ -0,0 +1,42 @@
+pragma solidity ^0.4.18;
+
+import "../Crowdsale.sol";
+import "../../token/ERC20/ERC20.sol";
+import "../../math/SafeMath.sol";
+
+
+/**
+ * @title AllowanceCrowdsale
+ * @dev Extension of Crowdsale where tokens are held by a wallet, which approves an allowance to the crowdsale.
+ */
+contract AllowanceCrowdsale is Crowdsale {
+  using SafeMath for uint256;
+
+  address public tokenWallet;
+
+  /**
+   * @dev Constructor, takes token wallet address. 
+   * @param _tokenWallet Address holding the tokens, which has approved allowance to the crowdsale
+   */
+  function AllowanceCrowdsale(address _tokenWallet) public {
+    require(_tokenWallet != address(0));
+    tokenWallet = _tokenWallet;
+  }
+
+  /**
+   * @dev Checks the amount of tokens left in the allowance.
+   * @return Amount of tokens left in the allowance
+   */
+  function remainingTokens() public view returns (uint256) {
+    return token.allowance(tokenWallet, this);
+  }
+
+  /**
+   * @dev Overrides parent behavior by transferring tokens from wallet.
+   * @param _beneficiary Token purchaser
+   * @param _tokenAmount Amount of tokens purchased
+   */
+  function _emitTokens(address _beneficiary, uint256 _tokenAmount) internal {
```

Comment:
I'm not super happy with this name

Reply:
discussed offline with @fiiiu and @facuspagnuolo and decided to go with `_deliverTokens` :pizza: